src/util/schnorr.rs

```diff
@@ -45,29 +45,59 @@ impl fmt::Display for TweakedPublicKey {
     }
 }
 
-/// A trait for tweaking Schnorr public keys
+/// Untweaked BIP-340 key pair
+pub type UntweakedKeyPair = KeyPair;
+
+/// Tweaked BIP-340 key pair
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+// TODO: Add other derives once secp256k1 v0.21.3 released
+pub struct TweakedKeyPair(KeyPair);
+
+/// A trait for tweaking BIP340 key types (x-only public keys and key pairs).
 pub trait TapTweak {
```

Comment:
As far as I understand this is an extension trait, so per Rust convention it should be called `KeyExt` or something. (Yes, it's a bit strange that it's an extension of two types actually.)

Reply:
I think extension traits are only about extending other *traits*.

Reply:
Not really, they are very often about extending types. [Even in `std`](https://doc.rust-lang.org/std/os/unix/ffi/trait.OsStrExt.html)

Reply:
This looks like a single-method extensions, which are usually named after the method name. The second method ("dangerous" one) is dangerous and I think changing the trait name will be more confusing than leaving as is ("KeyExt - extension to what? Which key types? Or extended keys?" And there may be other extenders in the future...).

Reply:
Yeah, true neither name is great.